ui/src/app/dashboard/views/DiscoveriesList/DiscoveriesList.tsx

```diff
@@ -1,16 +1,133 @@
+import { useEffect } from "react";
+
+import { MainTable, Tooltip, Icon } from "@canonical/react-components";
+import { useSelector, useDispatch } from "react-redux";
+
 import DiscoveriesListHeader from "../DiscoveriesListHeader";
 
 import Section from "app/base/components/Section";
 import { useWindowTitle } from "app/base/hooks";
+import { actions } from "app/store/discovery";
+import discoverySelectors from "app/store/discovery/selectors";
 
 const DiscoveriesList = (): JSX.Element => {
   useWindowTitle("Dashboard");
 
+  const discoveries = useSelector(discoverySelectors.all);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(actions.fetch());
+  }, [dispatch]);
+
+  const headers = [
+    {
+      content: "Name",
+      sortKey: "hostname",
+      className: "p-table--network-discoveries__name",
+    },
+    {
+      content: "Mac address",
+      sortKey: "mac_address",
+      className: "p-table--network-discoveries__mac",
+    },
+    {
+      content: "IP",
+      sortKey: "ip",
+      className: "p-table--network-discoveries__ip",
+    },
+    {
+      content: "Rack",
+      sortKey: "rack",
+      className: "p-table--network-discoveries__rack",
+    },
+    {
+      content: "Last seen",
+      sortKey: "last_seen",
```

Comment:
The default sorting handled by `MainTable` is very basic so it's not really working as intended here. The api returns a string, so it's sorting by string value meaning all the Mondays are getting grouped together, all the Tuesdays etc. instead of actually being sorted by time. The angular version also has this problem.

For now I would probably remove the ability to sort by `last_seen` (so just remove the `sortKey` and `sortData` for this column) and we can fix this as a follow up, either by parsing the string using `date-fns` or requesting that the api include a unix timestamp in the discovery model.